docs/recipes/Mocking-Api-Calls.md

```diff
@@ -2,35 +2,123 @@
+
+3. Set up the `dev-middleware.js` and `dev.endpoints.js`.
+  
+    ### `dev-middleware.js`
+  
+   A `dev-middleware.js` file allows you to setup a custom middleware configuration. This file is created by default when [One App module generator](https://github.com/americanexpress/one-app-cli/tree/master/packages/generator-one-app-module) is used in generating the module. Ensure that [`parrot-middleware`](https://github.com/americanexpress/parrot/tree/master/packages/parrot-middleware) is installed.
+
+   **Running within One App.**
+
+   Use the `set-middleware` command to link your module's custom dev middleware file to One App. Navigate to your local One App directory and run the following command:
+
+    ```bash
+    npm run set-middleware [path-to-your-module]/dev.middleware.js
+    ```
+
+   **Using One App runner**
+
+    When using [one-app-runner](https://github.com/americanexpress/one-app-cli/tree/master/packages/one-app-runner), this can be setup within the package.json as follows:
+  
+    ```json
+    "one-amex": {
+      "runner": {
+        "parrotMiddleware": "[path-to-your-module]/dev.middleware.js"
+      },
+    }
+    ```
+
+    The file `dev-middleware.js` contains the following parrot configurations.
+
+    ```javascript
+    const parrot = require('parrot-middleware');
+    const scenarios = require('./mock/scenarios');
+
+    module.exports = (app) => app.use(parrot(scenarios));
+    ```
+
+    ### `set-dev-endpoints.js`
+
+   A `dev.endpoints.js` file contains all the information One App needs to configure [one-app-dev-proxy](https://github.com/americanexpress/one-app-dev-proxy)
+   (our reverse proxy and mocking server that runs during development) and can be used to set remote endpoints
+   for your Module to use during local development. Create this file within the root folder of your module.
+
+   **Runng within One App.**
+
+   Use the `set-dev-endpoints` command to link your module's dev endpoints file to One App. Navigate to your local One App directory and run the following command: 
+
+   ```bash
+    npm run set-dev-endpoints [path-to-your-module]/dev.endpoints.js
+   ```
+
+   **Using One App runner**
+
+   When using [one-app-runner](https://github.com/americanexpress/one-app-cli/tree/master/packages/one-app-runner), this can be setup within the package.json as follows:
+  
+    ```json
+    "one-amex": {
+      "runner": {
+        "devEndpoints": "[path-to-your-module]/dev.endpoints.js"
+      },
+    }
+    ```
+
+   A `dev.endpoints.js` file contains the following:
+
+   ```js
+   module.exports = () => ({
+     apiUrl: {
+       devProxyPath: 'someApiUrl',
+       destination: 'https://example.com',
+     },
+   });
+    ```
+
+4. Start One App using the below command, this will allow the application to start with `one-app-dev-proxy` custom configurations setup above.
```

Comment:
if using one-app you will need to use the ` --use-middleware` flag when starting one-app